controllers/gce/controller/utils.go

```diff
@@ -289,6 +304,17 @@ func ListAll(store cache.Store, selector labels.Selector, appendFn cache.AppendF
 func (s *StoreToIngressLister) List() (ing extensions.IngressList, err error) {
```

Comment:
let us refactor here a little.

Refactor the current `StoreToIngressLister.List()` function into `ListGCEIngress()` and `ListGCEMultiClusterIngress`.

Change the return value from `extensions.IngressList` to `[]*extensions.Ingress`.

Change the input value for `GCETranslator.toNodePorts()` to use `[]*extensions.Ingress`

Reply:
List is also used at other places. I dont want all callers to get gce and gceMulti ingresses first and then append, so retaining the List method.

ListGCEMultiIngresses will not be used anywhere so not required atm.

Reply:
`List` is also used in `ListRuntimeInfo`. Then `lbs` is passed into `Checkpoint`. Then you filter lbs to singleClusterLbs in Checkpoint. 

Please just use `ListGCEIngresses` in ListRuntimeInfo and remove the filtering.

Reply:
I would prefer changing List to something else. Like `ListAll`. Next time, when someone tries to use it, it would not get confused and leave a bug that only triggers when there are a combination of gce and multi-cluster ingress.

Also, correct the comment to explicitly say GCE ingress and multi-cluster ingress

Reply:
Good catch.
Renamed List to ListAll and updated the comment to be explicit.
Also renamed ListRuntimeInfo to ListGCERuntimeInfo and added a comment stating that it returns runtimeinfo only for gce ingresses and not for multi cluster ingresses.
Also removed the IsMultiCluster field from RuntimeInfo and updated Checkpoint accordingly.